lib/matplotlib/colorbar.py

```diff
@@ -854,10 +857,22 @@ def get_ticks(self, minor=False):
             return self._long_axis().get_majorticklocs()
 
     @_api.delete_parameter("3.5", "update_ticks")
-    def set_ticklabels(self, ticklabels, update_ticks=True):
+    def set_ticklabels(self, ticklabels, *, minor=False, update_ticks=True,
+                       **kwargs):
         """
         Set tick labels.
 
+        .. admonition:: Discouraged
+
+            The use of this method is discouraged, because of the dependency
+            on tick positions. In most cases, you'll want to use
+            ``set_ticks(positions, labels)`` instead.
```

Comment:
This actually won't work as positional, because `update_ticks` is there in the way of the order.
```suggestion
            ``set_ticks(positions, labels=labels)`` instead.
```

Reply:
oops, thats true as well..